README.md

```diff
@@ -1 +1,35 @@
-# tss-ecdsa ![Build Status](https://github.com/novifinancial/tss-ecdsa/workflows/Rust%20CI/badge.svg)
+# tss-ecdsa
+
+This repo is a work-in-progress implementation of the ECDSA signing protocol described in "UC Non-Interactive, Proactive, Threshold ECDSA with Identifiable Aborts" [CGGMP] which can be found at https://eprint.iacr.org/2021/060.pdf.
+
+Specifically, we are targeting the three-round presigning protocol (with quadratic overhead for identifying faulty actors).
+
+This codebase is generally intended to be network-agnostic. Programs take messages as input and potentially output some outgoing messages in response. The relaying of these messages is assumed to happen externally. However, a proof-of-concept example of such networking code can be found in examples/network.
+
+##  What's Implemented
+
+### Key Generation (Figure 5 of CGGMP)
+
+KeyGen generates a threshold signing key, shares of which are distributed to each node. Every node outputs a private key along with the public keys of all other nodes. This only needs to be run once for a given set of nodes. 
+
+### Auxinfo (CGGMP Figure 6, minus the key refreshing)
+
+Auxinfo generates the auxilary information (such as Paillier keys) needed in order to compute presignatures. In CGGMP, this is done in parallel with key refreshing, however this codebase currently only implements the generation of auxilary information. This is run after KeyGen and only needs to be run once.
```

Comment:
Please be more specific as to what the Auxinfo contains.

Reply:
Question: Is key refreshing implemented in this codebase?

Reply:
It is not